push-to-s3-and-invalidate-cloudfront/action.yml

```diff
@@ -0,0 +1,78 @@
+name: Push files to an S3 bucket destination and invalidate the cloudfront cache
+description: |
+  This action will push files to a destination S3 bucket (or a folder within a bucket) and invalidates the cloudfront cache.
+  The action also removes any stale files from the bucket. The action can also delete files/folders from the bucket.
+
+inputs:
+  s3-bucket:
+    description: The name of the S3 bucket
+    required: true
+  build-directory:
+    description: The directory that contains the build files to upload to S3
+    required: true
+  aws-cloudfront-distribution-id:
+    description: The cloudfront distribution ID for the app
+    required: true
+  aws-region:
+    description: The aws region used by the IAM machine user
+    required: true
+  aws-access-key-id:
+    description: The aws access key id for the IAM machine user
+    required:  true
+  aws-secret-access-key:
+    description: The aws secret access key for the IAM machine user
+    required:  true

```

Comment:
Instead of inferring so much from the github context, I'd argue it is a lot better and clearer for the user to pass in a 'action' (ex: `create`, `update`, `delete`) to indicate what the user intends to do. 

Also the directory where the built files are put on the s3 bucket should not be inferred, it should be passed by the user. For example, pennylane has a different folder within production site for multiple different portions of the site built through different actions. It is not correct to assume "prod == / (root directory)". It is also very possible that callers of this action may not always want their PR builds structured in PR-{ID} but in different subfolders to schemas, and a general usecase action shouldn't enforce it either.

If we want this action to be general use, we need to make it flexible.

Reply:
Again, it doesn't have to be exactly like that above ^, but the idea is, more explicit, less implicit. And handle more usecases

Reply:
Thanks! Changes have been implemented